source/extensions/filters/http/grpc_stats/grpc_stats_filter.cc

```diff
@@ -207,6 +207,9 @@ class GrpcStatsFilter : public Http::PassThroughFilter {
     if (doStatTracking()) {
       config_->context_.chargeStat(*cluster_, Grpc::Context::Protocol::Grpc, request_names_,
                                    headers.GrpcStatus());
+      if (end_stream) {
+        maybeChargeUpstreamStat();
```

Comment:
Does this also need to be called in `encodeTrailers()`?

Reply:
I tested with only the code in the `encodeHeaders` and it only gave stats for the errors. Not sure why though, I didn't take some time to look into it. It won't work as is that's for sure

Reply:
? Isn't it already called in `encodeTrailers`?

Reply:
> Isn't it already called in encodeTrailers?

Yep, sorry; I didn't read carefully enough.